.github/workflows/build.yaml

```diff
@@ -42,29 +52,52 @@ jobs:
       with:
         default: '2.7.18'
 
-    - name: Cache kernel build
-      uses: actions/cache@v4
-      id: cache-kernel
+    - name: Restore cache kernel build
+      if: steps.cache-output.outputs.cache-hit != 'true'
+      uses: actions/cache/restore@v4
       with:
-        key: kernel-${{ steps.kernel-submodule.outputs.ref }}
-        path: |
-          agnos-kernel-sdm845/out
-        restore-keys: kernel-
+        key: kernel-out
+        path: agnos-kernel-sdm845/out
 
     - name: Build kernel
+      if: steps.cache-output.outputs.cache-hit != 'true'
```

Comment:
I don't like all these "skip if cache hit" since they add another case, and usually the time tradeoff isn't worth it.

Reply:
will remove them for now